### PR TITLE
hometask_5

### PR DESCRIPTION
**Pull request by izy-13**

### CODE REVIEW

File: docs/6-solid-principles/homework/homework.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Liskov Substitution Principle:
+https://github.com/13IEZ/UUI/blob/main/loveship/components/pickers/DataPickerBody.tsx абстракция над https://github.com/13IEZ/UUI/blob/main/uui-components/src/pickers/PickerBodyBase.tsx
+которая потом используется во всех пикерах loveship является LSP
+
+Interface Seggregation Principle:
+думаю, что большинство их компонентов разделяют данный принцип, https://github.com/13IEZ/UUI/blob/main/uui-components/src/pickers/PickerBodyBase.tsx
+потому что потом у каждого есть свой скин который дополняет base
+
+Dependency Inversion Principle:
```
Comment (AlenaBDE) [inline docs/6-solid-principles/homework/homework.txt:18]: This principle is not about the realisation of abstraction, is it about passing other class as dependency following the same interface, so you will substitute the dependency without broking code.

File: docs/6-solid-principles/homework/homework.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+возможно https://github.com/13IEZ/UUI/blob/develop/uui-core/hooks/useUuiServices.ts ? при добавлении нового сервиса всегда нужно будет ходить сюда и дополнять его
+
+Liskov Substitution Principle:
+https://github.com/13IEZ/UUI/blob/develop/uui-components/src/navigation/Anchor.tsx метод render переопределен 
+
+Interface Seggregation Principle:
+Возможно сюда можно отнести https://github.com/13IEZ/UUI/blob/develop/uui-components/src/buttons/Button.tsx строчки с 7 по 10
+
+Dependency Inversion Principle: 
+https://github.com/13IEZ/UUI/blob/develop/uui-components/src/buttons/ButtonBase.tsx возможно можно отнести 4 строчку но наверное больше смущает, что на 11 строке мы экстендим
```
Comment (AlenaBDE) [inline docs/6-solid-principles/homework/homework.txt:35]: i did not understand, why creation of new type is dependency inversion? and 11 line is empty

File: docs/6-solid-principles/homework/homework.txt (hunk 1 above)
Comment (izy-13) [reply docs/6-solid-principles/homework/homework.txt:18]: I'm out of ideas. I'm already going crazy when I look for this implementation in the repository where I started. I try to show it with my code from hometask_6. Let's imagine that:

our abstraction interface
interface WeightedGraph<T> {
  addVertex(vertex: T): void;
  addEdge(vertex1: T, vertex2: T, weight: number): void;
  getVertices(): Map<T, Map<T, number>>;
}

first implementation
class WeightedGraphImpl<T> implements WeightedGraph<T> {
  vertices: Map<T, Map<T, number>> = new Map();

  addVertex(vertex: T): void { some code }

  addEdge(vertex1: T, vertex2: T, weight: number): void { some code }

  getVertices(): Map<T, Map<T, number>> { some code }
}

second implementation
class NotWeightedGraphImpl<T> implements WeightedGraph<T> {
  vertices: Map<T, Map<T, number>> = new Map();

  addVertex(vertex: T): void { some code }

  addEdge(vertex1: T, vertex2: T, weight: number): void { some code }

  getVertices(): Map<T, Map<T, number>> { some code }
}

than class which depends on Interface
class DijkstraImpl  {
  private readonly graph: WeightedGraph<T>;

  constructor(graph: WeightedGraph<T>) {
    this.graph = graph;
  }
}

than we can easy change implements of WeightedGraph<T>

File: docs/6-solid-principles/homework/homework.txt (hunk 2 above)
Comment (izy-13) [reply docs/6-solid-principles/homework/homework.txt:35]: Can I say that I could implement DIP at hometask_6 and consider it bad DIP practice :) or if:

use implement nor interface
class DijkstraImpl {
private readonly graph: WeightedGraphImpl ;

constructor(graph: WeightedGraphImpl ) {
this.graph = graph;
}
}

File: docs/6-solid-principles/homework/homework.txt (hunk 1 above)
Comment (AlenaBDE) [reply docs/6-solid-principles/homework/homework.txt:18]: right, good example

**Review by AlenaBDE — APPROVED** [review]

**Pull request merged**